libsoftwarecontainer/src/gateway/devicenodegateway.cpp

```diff
@@ -62,11 +75,11 @@ bool DeviceNodeGateway::activateGateway()
 {
     for (auto &dev : m_devList) {
         log_info() << "Mapping device " << dev.name;
-        const int mode = std::atoi(dev.mode.c_str());
 
         if (dev.major.length() != 0) {
             const int majorVersion = std::atoi(dev.major.c_str());
             const int minorVersion = std::atoi(dev.minor.c_str());
+            const int mode = std::atoi(dev.mode.c_str());
```

Comment:
Solved this by using parseInt that is already used in the agent so moved it to comon.